charts/csi-secret-provider-class/templates/deployment.yaml

```
@@ -0,0 +1,45 @@
+{{- if .Values.dummyDeployment.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+{{ template "common.metadata" . }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "csi-secret-provider-class.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+    {{- with .Values.dummyDeployment.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      labels:
+        {{- include "csi-secret-provider-class.selectorLabels" . | nindent 8 }}
+    spec:
+      containers:
+        - name: dummy
+          image: {{ .Values.dummyDeployment.image.repository }}:{{ .Values.dummyDeployment.image.tag }}
+          imagePullPolicy: {{ .Values.dummyDeployment.image.pullPolicy }}
+          {{- with .Values.dummyDeployment.resources }}
+          resources:
+            {{- toYaml .Values.dummyDeployment.resources | nindent 12 }}
+          {{- end }}
```

Use `.` when using `with`.
```suggestion
          {{- with .Values.dummyDeployment.resources }}
          resources:
            {{- toYaml .| nindent 12 }}
          {{- end }}
```